src/components/SubmissionConfirmation.js

```diff
@@ -1,75 +1,160 @@
-import React from 'react';
+import React, {useState} from 'react';
 import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+import useAsync from 'react-use/esm/useAsync';
 
+import { post } from 'api';
 import Body from 'components/Body';
 import Card from 'components/Card';
 import FAIcon from 'components/FAIcon';
 import Anchor from 'components/Anchor';
 import Loader from 'components/Loader';
+import PaymentForm from 'components/PaymentForm';
 import usePoll from 'hooks/usePoll';
 
+const RESULT_FAILED = 'failed';
+
+const getStartPaymentUrl = (apiUrl) => {
+  const nextUrl = new URL(window.location.href);
+  const startPaymentUrl = new URL(apiUrl);
+  startPaymentUrl.searchParams.set("next", nextUrl.toString());
+  return startPaymentUrl.toString();
+};
+
+/**
+ * Given a start URL, fetches the payment start information and renders the UI controls
+ * to enter the payment flow.
+ * @param  {String} startUrl The payment start URL received from the backend.
+ * @return {JSX}
+ */
+const StartPayment = ({startUrl}) => {
+  const fullUrl = getStartPaymentUrl(startUrl);
+  const {loading, value} = useAsync(
+    async () => {
+      const resp = await post(fullUrl);
```

Comment:
At the moment if this request throws an error, it will break the page and the user might not be able to read their submission reference. Maybe good to have a try/catch that then displays an error?

Reply:
Good catch - decided to wrap it in an `ErrorBoundary` and tested by manually `throw`ing in the async callback.